qiskit/circuit/qpy_serialization.py

```diff
@@ -847,14 +840,16 @@ def _write_circuit(file_obj, circuit):
     clbit_indices = {bit: index for index, bit in enumerate(circuit.clbits)}
     if num_registers > 0:
         for reg in circuit.qregs:
+            standalone = reg[0]._register is not None
```

Comment:
```suggestion
            standalone = all(bit._register is reg for bit in reg)
```
might be necessary, because you could e.g.
```
qr_foo = QuantumRegister('foo', 5)
qr_foo = QuantumRegister('bar', bits=qr_foo[:3] + [Qubit(), Qubit()])
```

Alternately, we could tag registers on creation as being "standalone" or "composite/aliased/shared/...(?)", but that begs the question of why we want to keep this information on serialized registers. The `__repr__` of bits initialized through `QuantumRegister('foo', N)` and `Qubit()` differ, and they're drawn differently in the case where both are contained in a `Register`, but the latter is a bug. Am I missing something else?

Reply:
Oh, I hadn't thought about that I figured people would do just 1 or the other. Then yeah I'll switch it to be like this, I think the `__repr__` and drawer are the only differences I can think of, and this is enough of an edge case I doubt it'll come up.  If it ends up being a problem we can always add support for hybrid/composite/aliased/shared/.. in a v2 format spec.